src/hotspot/cpu/aarch64/c1_MacroAssembler_aarch64.cpp

```diff
@@ -321,9 +328,7 @@ void C1_MacroAssembler::allocate_array(Register obj, Register len, Register t1,
 void C1_MacroAssembler::inline_cache_check(Register receiver, Register iCache) {
   verify_oop(receiver);
   // explicit null check not needed since load from [klass_offset] causes a trap
-  // check against inline cache
-  assert(!MacroAssembler::needs_explicit_null_check(oopDesc::klass_offset_in_bytes()), "must add explicit null check");
```

Comment:
We had better not remove this but use the condition !UseCompactObjectHeaders

Reply:
done